blocks/basic/include/gnuradio-4.0/basic/StreamToDataSet.hpp

```diff
@@ -0,0 +1,227 @@
+
+    // port definitions
+    using OutType = std::conditional_t<streamOut, T, DataSet<T>>;
+    PortIn<T>               in;
+    PortOut<OutType, Async> out;
+
+    // settings
+    A<std::string, "filter", Visible, Doc<"syntax: '[<start trigger name>/<ctx1>, <stop trigger name>/<ctx2>]'">> filter;
+    A<property_map, "filter state", Doc<"">>                                                                      filterState;
+    A<gr::Size_t, "n samples pre", Visible, Doc<"number of pre-trigger samples">>                                 n_pre  = 0U;
+    A<gr::Size_t, "n samples post", Visible, Doc<"number of post-trigger samples">>                               n_post = 0U;
+    A<gr::Size_t, "n samples max", Doc<"maximum number of samples (0: infinite)">>                                n_max  = 0U;
+
+    // meta information (will be usually set by incoming tags/upstream sources
+    A<float, "sample_rate", Doc<"signal sample rate">>                                                       sample_rate = 1.f;
+    A<std::string, "signal_name", Doc<"signal name">>                                                        signal_name;
+    A<std::string, "signal quantity", Doc<"physical quantity (e.g., 'voltage'). Follows ISO 80000-1:2022.">> signal_quantity;
+    A<std::string, "signal unit", Doc<"unit of measurement (e.g., '[V]', '[m]'). Follows ISO 80000-1:2022">> signal_unit;
+    A<float, "signal_min", Doc<"signal physical max. (e.g. DAQ) limit">>                                     signal_min = 0.f;
+    A<float, "signal_max", Doc<"signal physical max. (e.g. DAQ) limit">>                                     signal_max = 1.f;
+
+    // internal trigger state
+    HistoryBuffer<T> _history{ MIN_BUFFER_SIZE + n_pre };
+    TMatcher         _matcher{};
+    bool             _accumulationActive  = false;
+    std::size_t      _nSamplesWritten     = 0UZ;
+    std::size_t      _nPostSamplesWritten = 0UZ;
+    DataSet<T>       _tempDataSet;
+    bool             _dataSetAccumulation = false;
+
+    void
+    reset() {
+        filterState.value.clear();
+        _accumulationActive  = false;
+        _dataSetAccumulation = false;
+    }
+
+    void
+    settingsChanged(const gr::property_map & /*oldSettings*/, const gr::property_map &newSettings) {
+        if (newSettings.contains("n_pre")) {
+            auto newBuffer = HistoryBuffer<T>(MIN_BUFFER_SIZE + std::get<gr::Size_t>(newSettings.at("n_pre")));
+            newBuffer.push_back_bulk(_history);
+            _history = std::move(newBuffer);
+        }
+    }
+
+    gr::work::Status
+    processBulk(ConsumableSpan auto inSamples /* equivalent to std::span<const T> */, PublishableSpan auto &outSamples /* equivalent to std::span<T> */) {
+        bool firstTrigger = false;
+        bool lastTrigger  = false;
+        if (const trigger::MatchResult matchResult = _matcher(filter.value, this->mergedInputTag(), filterState.value); matchResult != trigger::MatchResult::Ignore) {
```

Comment:
In this case -- since there are no other stride or `[input, output]_chunck_size` nor ports defined -- the `mergedInputTag()` is just merging tags that have been placed on the same sample. That should be safe.

Reply:
In this case -- since there are no other stride or `[input, output]_chunk_size` constraints nor additional input ports -- there cannot be missed triggers unless the user places them in the same sample.

We can revamp this once the new tag API is in place.